.circle/bintray.sh

```diff
@@ -143,24 +139,30 @@ function check_package_exists() {
   return ${package_exists}
 }
 
-function create_package() {
-  debug "Creating package ${PKG_NAME}..."
-  data="{
-    \"name\": \"${PKG_NAME}\",
-    \"desc\": \"Packages for StackStorm event-driven automation platform\",
-    \"vcs_url\": \"https://github.com/stackstorm/st2.git\",
-    \"licenses\": [\"Apache-2.0\"],
-    \"labels\": [\"st2\", \"StackStorm\", \"DevOps\", \"automation\", \"auto-remediation\", \"chatops\"],
-    \"website_url\": \"https://stackstorm.com\",
-    \"issue_tracker_url\": \"https://github.com/stackstorm/st2/issues\",
-    \"github_repo\": \"stackstorm/st2\",
-    \"github_release_notes_file\": \"CHANGELOG.rst\",
-    \"public_download_numbers\": false,
-    \"public_stats\": true
-  }"
-
-  ${CURL} -X POST -d "${data}" ${API}/packages/${BINTRAY_ORGANIZATION}/${BINTRAY_REPO}/
-  echo ""
+function ensure_package() {
+  # The .bintray_package file contains package settings. You can use env variables in the file.
+  # The file should be located in CWD (st2-packages repo for st2, st2web for web ui and so on).
+  # See https://bintray.com/docs/api/#_create_package for format and requirements.
+  if [[ ! -f .bintray_package ]] ; then
+    echo 'File ".bintray_package" is not there, aborting.'
+    exit
+  fi
+  data=$(eval "cat <<EOF
+$(<.bintray_package)
+EOF" 2> /dev/null)
+
+  if (! check_package_exists); then
+    debug "The package ${PKG_NAME} does not exist"
+    debug "Creating package ${PKG_NAME}..."
+
+    ${CURL} -X POST -d "${data}" ${API}/packages/${BINTRAY_ORGANIZATION}/${BINTRAY_REPO}/
+    echo ""
+  else
+    debug "Updating package ${PKG_NAME}..."
+
+    ${CURL} -X PATCH -d "${data}" ${API}/packages/${BINTRAY_ORGANIZATION}/${BINTRAY_REPO}/${PKG_NAME}/
```

Comment:
> Why should we avoid editing packages in Bintray.com manually via Web ui where it's really needed?

For one, because we have a packet per distro and no way of automate the change. That's A LOT of manual changing. Last time I've checked, a single change in current `.bintray_package` inside st2-packages will require you to change it for every single one of 60+ packages out there.

But mostly, it's just a benefits of `* as a code` - known state, history, audit and all that jazz.